apollo-ios/Sources/Apollo/GraphQLExecutor.swift

```diff
@@ -191,10 +197,51 @@ final class GraphQLExecutor<Source: GraphQLExecutionSource> {
     variables: GraphQLOperation.Variables? = nil,
     accumulator: Accumulator
   ) throws -> Accumulator.FinalResult {
-    let info = ObjectExecutionInfo(
-      rootType: SelectionSet.self,
+    return try execute(
+      selectionSet: selectionSet,
+      on: data,
       variables: variables,
+      withRootCacheReference: root,
       schema: SelectionSet.Schema.self,
+      accumulator: accumulator
+    )
+  }
+
+  func execute<
+    Accumulator: GraphQLResultAccumulator,
+    Operation: GraphQLOperation
+  >(
+    selectionSet: any SelectionSet.Type,
+    in operation: Operation.Type,
+    on data: Source.RawObjectData,
+    withRootCacheReference root: CacheReference? = nil,
+    variables: GraphQLOperation.Variables? = nil,
+    accumulator: Accumulator
+  ) throws -> Accumulator.FinalResult {
+    return try execute(
+      selectionSet: selectionSet,
+      on: data,
+      variables: variables,
+      withRootCacheReference: root,
+      schema: Operation.Data.Schema.self,
+      accumulator: accumulator
+    )
+  }
+
+  private func execute<
+    Accumulator: GraphQLResultAccumulator
+  >(
+    selectionSet: any SelectionSet.Type,
+    on data: Source.RawObjectData,
+    variables: GraphQLOperation.Variables? = nil,
```

Comment:
variables should come after withRootCacheReference to mirror above methods.

Reply:
Done in Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/a92e0e6f57e8c02b74a105ba9d0fb13e2d16c70e..